core/src/integration-test/java/io/aiven/kafka/tieredstorage/RemoteStorageManagerTest.java

```diff
@@ -479,16 +478,15 @@ void testTransformingIndexes(final boolean encryption) {
             "storage.root", targetDir.toString(),
             "encryption.enabled", Boolean.toString(encryption)
         ));
-        final SegmentEncryptionMetadataV1 encryptionMetadata;
+        final DataKeyAndAAD maybeEncryptionKey;
```

Comment:
What about using an `Optional` instead of setting `maybeEncryptionKey` to `null` ?
Then when using `maybeEncryptionKey` you don't need to check again if `encryptionEnabled`, but if the value is present in the `Optional` then it means that encryption is enabled and you will use that encryption key. Probably it does not reduce the code but I find it more easy to reason about explicit nulls.